src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -130,11 +130,11 @@
 
 (def required-master-files
   "The set of SSL files that are required on the master."
-  [:hostprivkey :hostcert])
+  #{:hostprivkey :hostcert :localcacert})
 
 (def required-ca-files
   "The set of SSL related files that are required on the CA."
-  [:cacert :cacrl :cakey :cert-inventory :serial])
+  #{:cacert :cacrl :cakey :cert-inventory :serial})
```

Comment:
I had been expecting we would be adding `:localcacert` here instead of to `required-master-files`, so that the CA certificate artifacts don't get created if the `:localcacert` already exists.  This might be the case for someone seeking to do an external CA setup for their master who forgot to replace the real certificate-authority service with the "disabled" certficate-authority service before restarting the master.